### PR TITLE
[MED-21718] Fix rollback to retired release

### PR DESCRIPTION
**Pull request by matkam**
Hello @dokipen, @ddbenson, @dnelson, @silverlyra, @micahnoland, 

Please review the commits I made in branch 'matkam/fix-rollback-to-retired'.

R=@dokipen
R=@ddbenson
R=@dnelson
R=@silverlyra
R=@micahnoland

### CODE REVIEW

File: pkg/controller/releasemanager/incarnation.go [hunk 1]
```diff
@@ -857,10 +856,14 @@ func (i *IncarnationCollection) releasable() (r []*Incarnation) {
 		}
 	}
 	if len(r) == 0 && len(i.willRelease()) == 0 {
-		i.controller.getLog().Info("there are no releases, looking for retired release to unretire")
+		i.controller.getLog().Info("no viable releases. looking for releases to take out of retirement")
 		candidates := i.unretirable()
 		if len(candidates) > 0 {
-			candidates[0].fastRelease()
+			firstUnretireable := candidates[0]
```
Comment (dokipen) [inline pkg/controller/releasemanager/incarnation.go:862]: I'm confused about this. Shouldn't fast release make it so it transitions to deploying on the next reconcile loop? Do you understand why this fixes it?

File: pkg/controller/releasemanager/syncer_test.go [hunk 2]
```diff
@@ -316,3 +316,30 @@ func TestPrepareRevisionsAndRulesIncompleteScaleUp(t *tt.T) {
...
+		log:          test.MustNewLogger(),
+	}
+
+	releasedIncarnations := []*Incarnation{
+		createTestIncarnation("latest-good", retired, 0),
+		//createTestIncarnation("latest-failing", failing, 100), // failing incarnations do not get returned by releasable()
+		createTestIncarnation("old", retired, 0),
+	}
+	releasedIncarnations[0].fastRelease()
+	releasedIncarnations[0].status.Scale.Current = 1
```
Comment (silverlyra) [inline pkg/controller/releasemanager/syncer_test.go:336]: why do these get set here in the test? a `retired` revision would never have a nonzero current scale, right?

File: pkg/controller/releasemanager/incarnation.go (hunk 1 above)
Comment (matkam) [reply pkg/controller/releasemanager/incarnation.go:862]: I have not tested this yet. I'm planning on running it in sandbox today to see if it actually works. But the problem is that setting `"releaseEligible": true` does not actually get saved on retired releases, so they never transition to `deploying`.

File: pkg/controller/releasemanager/syncer_test.go (hunk 2 above)
Comment (matkam) [reply pkg/controller/releasemanager/syncer_test.go:336]: You're right about the scale. I should not be setting this here. `fastRelease()` does happen though, when there are no other good releases.

**Pull request merged**